### ISSUE TITLE
10g port is not coming up in S5148

### ISSUE THREAD

**Issue by waliulislam** [open-switch/opx-platform-config#50]
In S5148 , 25g access port has 10g SFPPLUS transceiver and connected to s4248 10 g port. Link is not coming up. I have tried both autoneg "on" and "off" settings in copper and fiber cable.

root@TR1:/etc/opx# opx-ethtool e101-017-0
Settings for e101-017-0:
    Channel ID:   0
    Transceiver Status: Enable
    Media Type: SFPPLUS 10GBASE SR
    Part Number: FTLX8571D3BCL
    Serial Number: APL1HNJ
    Qualified: Yes
    Administrative State: UP
    Operational State: DOWN
    Supported Speed (in Mbps):  [25000, 10000]
    Auto Negotiation : off
    Configured Speed   : 10000
    Operating Speed   : 0
    Duplex   : full
root@TR1:/etc/opx#

root@TR1:/etc/opx# opx-show-transceivers 17
Front Panel Port 17 media details
==================================
Key: 2.28.1835221.1835077.1835078.1835079.
base-pas/media/rate-identifier = 0
base-pas/media/voltage-low-warning-threshold = 3.0
base-pas/media/cc_ext = 244
base-pas/media/media-category/qsfp-plus/wavelength-tolerance = 0
base-pas/media/vendor-id = 009065
base-pas/media/sub-port-id = 17
base-pas/media/voltage-state = 1
base-pas/media/port-type = 1
base-pas/media/bias-low-alarm-threshold = 4.0
base-pas/media/default-breakout-speed = 4
base-pas/media/media-name = SFP+ 10GBASE SR
base-pas/media/temp-high-warning-threshold = 73.0
base-pas/media/cable-length-cm = 0
base-pas/media/bias-high-alarm-threshold = 13.1999998093
base-pas/media/options = 26
base-pas/media/tx-power-high-warning-threshold = -1.00015425682
base-pas/media/slot = 1
base-pas/media/port = 17
base-pas/media/connector-separable = 1
base-pas/media/temp-low-warning-threshold = -8.0
base-pas/media/default-autoneg = 3
base-pas/media/enhanced-options = 240
base-pas/media/oper-status = 0
base-pas/media/category = 2
base-pas/media/port = 17
base-pas/media/bias-high-warning-threshold = 12.6000003815
base-pas/media/insertion-cnt = 1
base-pas/media/qsa28-expected = 0
base-pas/media/admin-status = 0
base-pas/media/slot = 1
base-pas/media/default-breakout-mode = 11
base-pas/media/encoding = 6
base-pas/media/category-string = SFP+
base-pas/media/capability = 4
base-pas/media/diag-mon-type = 104
base-pas/media/channel-count = 1
base-pas/media/type = 5
base-pas/media/temp-high-alarm-threshold = 78.0
base-pas/media/length-sfm = 0
base-pas/media/serial-number = APL1HNJ
base-pas/media/temp-low-alarm-threshold = -13.0
base-pas/media/tx-power-low-warning-threshold = -5.00038146973
base-pas/media/insertion-timestamp = 140637088816208
base-pas/media/high-power-mode = 0
base-pas/media/br-nominal = 103
base-pas/media/media-interface-qualifier = 2
base-pas/media/rx-power-high-warning-threshold = -1.00015425682
base-pas/media/default-phy-mode = 1
base-pas/media/voltage-high-warning-threshold = 3.59999990463
base-pas/media/tx-power-high-alarm-threshold = 0.0
base-pas/media/display-string = SFP+ 10GBASE SR
base-pas/media/vendor-pn = FTLX8571D3BCL
base-pas/media/current-temperature = 23.40234375
base-pas/media/length-om2 = 8
base-pas/media/length-om3 = 30
base-pas/media/rx-power-low-alarm-threshold = -20.0
base-pas/media/length-om1 = 3
base-pas/media/media-category/sfp-plus/br-max = 0
base-pas/media/connector = 7
base-pas/media/lockdown-state = 1
base-pas/media/vendor-name = FINISAR CORP.
base-pas/media/vendor-rev = 4100
base-pas/media/media-category/sfp-plus/sff-8472-compliance = 0
base-pas/media/qualified = 1
base-pas/media/media-interface = 9
base-pas/media/wavelength = 850
base-pas/media/support-status = 0
base-pas/media/date-code = 3133303531370000
base-pas/media/length-cable = 0
base-pas/media/identifier = 3
base-pas/media/media-interface-lane-count = 1
base-pas/media/length-sfm-km = 0
base-pas/media/rx-power-high-alarm-threshold = 0.0
base-pas/media/media-category/sfp-plus/br-min = 0
base-pas/media/rx-power-measurement-type = 104
base-pas/media/bias-low-warning-threshold = 5.0
base-pas/media/rate-select-state = 0
base-pas/media/media-category/qsfp-plus/max-case-temp = 0
base-pas/media/ext-transceiver = 0
base-pas/media/tx-power-low-alarm-threshold = -5.99980401993
base-pas/media/rx-power-low-warning-threshold = -18.0134296417
base-pas/media/temp-state = 1
base-pas/media/ext-identifier = 4
base-pas/media/cc_base = 72
base-pas/media/qsa-adapter = 2
base-pas/media/ext-spec-compliance-code = 0
base-pas/media/connector-type = 2
base-pas/media/present = 1
base-pas/media/transceiver = 0100000000000000
base-pas/media/port-density = 1
base-pas/media/voltage-high-alarm-threshold = 3.70000004768
base-pas/media/voltage-low-alarm-threshold = 2.90000009537
base-pas/media/default-media-speed = 4
base-pas/media/current-voltage = 3.33229994774
base-pas/media/cable-type = 2
Key: 2.28.1835222.1835020.1835021.1835022.1835023.
base-pas/media-channel/port = 17
base-pas/media-channel/tx-power = -1.91721487045
base-pas/media-channel/rx-loss = 0
base-pas/media-channel/tx-power-state = 1
base-pas/media-channel/channel = 0
base-pas/media-channel/oper-status = 0
base-pas/media-channel/cdr-enable = 0
base-pas/media-channel/tx-fault = 0
base-pas/media-channel/tx-loss = 0
base-pas/media-channel/port = 17
base-pas/media-channel/slot = 1
base-pas/media-channel/channel = 0
base-pas/media-channel/tx-disable = 0
base-pas/media-channel/rx-power-state = 1
base-pas/media-channel/rx-power = -2.75642204285
base-pas/media-channel/state = 1
base-pas/media-channel/tx-bias-state = 1
base-pas/media-channel/tx-bias-current = 7.48199987411
base-pas/media-channel/slot = 1
base-pas/media-channel/speed = 4
root@TR1:/etc/opx#

I have tried 40g copper transceiver in 100g port and fanout to 10g  and also 100 gig fiber port and 100gig fiber port fanout to 25g. Link came up in these cases.

**Comment by jeff-yin** [comment]
Workaround is to manually configure the port's speed using `opx-ethtool`

**Comment by atanu-mandal** [comment]
This issue has been release noted in OPX 3.0.0 as 

> S5148-ON interface does not come up if 10G optic is plugged into a 25G port
For both copper/fiber modes, use the "opx-ethtool-s e101-017-0 autonet off speed 10000" command to bring up the port

**Comment by jeff-yin** [comment]
We will use this bug to track the various autoneg discrepancies between platforms

**Comment by atanu-mandal** [comment]
There was service dependency fix made in opx-nas-interface recently. Requested to verify having upgraded

**Comment by jeff-yin** [comment]
Various interface-related issues have since been fixed in the latest OPX 3.1.0 release. @waliulislam please verify when you get a chance
